Yatima/ToIpld.lean

```diff
@@ -85,11 +88,94 @@ def exprMetaToIpld : ExprMeta → Ipld
+  | .safe      => .array #[.number DEFINITIONSAFETY, .number 0]
+  | .«unsafe»  => .array #[.number DEFINITIONSAFETY, .number 1]
+  | .«partial» => .array #[.number DEFINITIONSAFETY, .number 2]
+
+def definitionAnonToIpld : DefinitionAnon -> Ipld
+  | .mk l t v s => .array #[.number DEFINITIONANON, .number 0, l, t, v, definitionSafetyToIpld s]
+
+def definitionMetaToIpld : DefinitionMeta -> Ipld
+  | .mk n l t v => .array #[.number DEFINITIONMETA, .number 0, n, l, t, v]
+
+def inductiveAnonToIpld : InductiveAnon -> Ipld
+  | .mk l t p i c r s rf n => .array #[.number INDUCTIVEANON, .number 0, l, t, p, i, c, r, s, rf, n]
+
+def inductiveMetaToIpld : InductiveMeta -> Ipld
+  | .mk n l t c => .array #[.number INDUCTIVEMETA, .number 0, n, l, t, c]
+
+def constructorAnonToIpld : ConstructorAnon -> Ipld
+  | .mk l t i idx p f s => .array #[.number CONSTRUCTORANON, .number 0, l, t, i, idx, p, f, s]
+
+def constructorMetaToIpld : ConstructorMeta -> Ipld
+  | .mk n l t i => .array #[.number CONSTRUCTORMETA, .number 0, n, l, t, i]
+
+def recursorRuleAnonToIpld : RecursorRuleAnon -> Ipld
+  | .mk c f r => .array #[.number RECURSORRULEANON, .number 0, c, f, r]
+
+instance : Coe RecursorRuleAnon Ipld where coe r := recursorRuleAnonToIpld r
+
+def recursorRuleMetaToIpld : RecursorRuleMeta -> Ipld
+  | .mk c r => .array #[.number RECURSORRULEMETA, .number 0, c, r]
+
+instance : Coe RecursorRuleMeta Ipld where coe r := recursorRuleMetaToIpld r
+
+def recursorAnonToIpld : RecursorAnon -> Ipld
+  | .mk l t i p is m mi r k s => .array #[.number RECURSORANON, .number 0, l, t, i, p, is, m, mi, r, k, s]
+
+def recursorMetaToIpld : RecursorMeta -> Ipld
+  | .mk n l t i r => .array #[.number RECURSORMETA, .number 0, n, l, t, i, r]
+
+def quotKindToIpld : QuotKind -> Ipld
+  | .type      => .array #[.number QUOTKIND, .number 0]
+  | .ctor      => .array #[.number QUOTKIND, .number 1]
+  | .lift      => .array #[.number QUOTKIND, .number 3]
+  | .ind       => .array #[.number QUOTKIND, .number 4]
+
+def quotientAnonToIpld : QuotientAnon -> Ipld
+  | .mk l t k => .array #[.number QUOTIENTANON, .number 0, l, t, quotKindToIpld k]
+
+def quotientMetaToIpld : QuotientMeta -> Ipld
+  | .mk n l t => .array #[.number QUOTIENTMETA, .number 0, n, l, t]
+
 def constAnonToIpld : ConstAnon → Ipld
-  | _ => sorry
+  | .«axiom» a     => .array #[.number CONSTANON, .number 0, axiomAnonToIpld a]
+  | .«theorem» t   => .array #[.number CONSTANON, .number 1, theoremAnonToIpld t]
+  | .«inductive» i => .array #[.number CONSTANON, .number 2, inductiveAnonToIpld i]
+  | .opaque o      => .array #[.number CONSTANON, .number 3, opaqueAnonToIpld o]
+  | .definition d  => .array #[.number CONSTANON, .number 4, definitionAnonToIpld d]
+  | .constructor c => .array #[.number CONSTANON, .number 5, constructorAnonToIpld c]
+  | .recursor r    => .array #[.number CONSTANON, .number 6, recursorAnonToIpld r]
+  | .quotient q    => .array #[.number CONSTANON, .number 7, quotientAnonToIpld q]
```

Comment:
I think you can go even further and completely replace the functions with direct instances.

Also, here's another idea: have, for example, a `Coe` instance from `AxiomAnon` to `Array Ipld`.

So instead of
```lean
def axiomAnonToIpld : AxiomAnon → Ipld
  | .mk l t s  => .array #[.number AXIOMANON, .number 0, l, t, s]
```
You'd have
```lean
instance Coe AxiomAnon (Array Ipld) where coe
  | .mk l t s => #[l, t, s]
```
And then instead of
```lean
| .«axiom» a => .array #[.number CONSTANON, .number 0, axiomAnonToIpld a]
```
You'd do
```lean
| .«axiom» a => .array #[.number CONSTANON, .number 0] ++ a
```

This would also eliminate the need for defining so many constants on `Yatima/Cid.lean`

Reply:
Done.